### DISCUSSION TITLE
Can Typer display the effect of having both a log and a progress bar like in APT?

### DISCUSSION THREAD

**Discussion by lee-cq** [fastapi/typer#709] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
null
```

### Description

My CLI program contains a lot of log output, it was working fine, now I have a new requirement:

I need to implement an encompassing progress bar at the bottom of the terminal to track the progress of a CLI program.

I found an implementation effect that I expected. Ubuntu's apt-get command is the effect I most expect when installing applications.

Can Typer achieve such an effect?

### Operating System

Linux

### Operating System Details

There is a log above it and a progress bar at the bottom.

![image](https://github.com/tiangolo/typer/assets/47050568/caad7989-5ad5-4cad-8467-3182b1b42928)

### Typer Version

0.9.0

### Python Version

Python 3.10.12

**Comment by The-Compiler** [comment, marked as answer]
Typer takes care of parsing CLI arguments, it's not concerned with how to do output while your application is running.

A popular library to do this kind of thing (which Typer even integrates with!) is [rich](https://github.com/textualize/rich/), see [Progress Display — Rich 13.6.0 documentation](https://rich.readthedocs.io/en/latest/progress.html)
